packages/smart-contract/src/network/mina-network.ts

```diff
@@ -18,14 +20,19 @@ export class MinaNetwork {
     return this.instance;
   }
 
-  public connect(networkId: NetworkId, endpoint: string) {
+  public connect(networkId: NetworkId, endpoint: string, apiKey: string) {
     Mina.setActiveInstance(
       Mina.Network({
         networkId,
         mina: endpoint,
       })
     );
 
+    this.#api = new BlockberryApi(
+      apiKey,
+      networkId === 'mainnet' ? 'mainnet' : 'devnet'
+    );
```

Comment:
no need to specify this

Reply:
we have mainnet and testnet NetworkId, but that endpoint requires devnet

Reply:
You mean to prevent conflict between devnet and testnet?

If that so you can check networkId === 'devnet' ? 'mainnet' : networkId